checks/pinned_dependencies_test.go

```diff
@@ -1038,10 +1198,80 @@ func TestGitHubWorkflowUsesLineNumber(t *testing.T) {
 			}
 			for _, expectedLog := range tt.expected {
 				isExpectedLog := func(logMessage checker.LogMessage, logType checker.DetailType) bool {
-					return logMessage.Offset == expectedLog.lineNumber && logMessage.Path == tt.filename &&
+					return logMessage.Offset == expectedLog.startLine &&
+						logMessage.EndOffset == expectedLog.endLine &&
+						logMessage.Path == tt.filename &&
 						logMessage.Snippet == expectedLog.dependency && logType == checker.DetailWarn &&
-						strings.Contains(logMessage.Text, "dependency not pinned by hash")
+						strings.Contains(logMessage.Text, "action not pinned by hash")
 				}
+
+				if !scut.ValidateLogMessage(isExpectedLog, &dl) {
+					t.Errorf("test failed: log message not present: %+v", tt.expected)
+				}
+			}
+		})
+	}
+}
+
+func TestGitHubWorkInsecureDownloadsLineNumber(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name     string
+		filename string
+		expected []struct {
+			snippet   string
+			startLine uint
+			endLine   uint
+		}
+	}{
+		{
+			name:     "downloads",
+			filename: "testdata/github-workflow-download-lines.yaml",
+			expected: []struct {
+				snippet   string
+				startLine uint
+				endLine   uint
+			}{
+				{
+					snippet:   "bash /tmp/file",
+					startLine: 27,
+					endLine:   2,
```

Comment:
That's weird to me that endLine is less than startLine. Why is that?

Reply:
will fix, slipped thru the cracks :-)